v2/apierror/apierror.go

```diff
@@ -60,6 +61,23 @@ type ErrDetails struct {
 	Unknown []interface{}
 }
 
+// ExtractMessage provides a mechanism for extracting messages from the Unknown
+// error details.  An entry matching the type of exampleMesg is returned
+// if present.
+func (e ErrDetails) ExtractMessage(exampleMesg interface{}) interface{} {
```

Comment:
Instead of `exampleMesg ` being a specification that is thrown away, what about accepting a pointer that is populated with the extracted value, similar to the `v` param in [json.Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal)?

Reply:
Swapped this to more closely mimic that pattern.